yb-voyager/cmd/importData.go

```diff
@@ -105,15 +80,76 @@ var importDataCmd = &cobra.Command{
 	PreRun: func(cmd *cobra.Command, args []string) {
 		validateImportFlags(cmd)
 	},
+	Run: importDataCommandFn,
+}
 
-	Run: func(cmd *cobra.Command, args []string) {
-		target.ImportMode = true
-		sourceDBType = ExtractMetaInfo(exportDir).SourceDBType
-		sqlname.SourceDBType = sourceDBType
-		dataStore = datastore.NewDataStore(filepath.Join(exportDir, "data"))
-		checkExportDataDoneFlag()
-		importData()
-	},
+func importDataCommandFn(cmd *cobra.Command, args []string) {
+	reportProgressInBytes = false
+	target.ImportMode = true
+	sourceDBType = ExtractMetaInfo(exportDir).SourceDBType
+	sqlname.SourceDBType = sourceDBType
+	dataStore = datastore.NewDataStore(filepath.Join(exportDir, "data"))
+	checkExportDataDoneFlag()
+	dataFileDescriptor = datafile.OpenDescriptor(exportDir)
+	importFileTasks := discoverFilesToImport()
+	importFileTasks = applyTableListFilter(importFileTasks)
+	importData(importFileTasks)
+}
+
+type ImportFileTask struct {
+	ID        int
+	FilePath  string
+	TableName string
+}
+
+func discoverFilesToImport() []*ImportFileTask {
+	result := []*ImportFileTask{}
+	exportDataDir := filepath.Join(exportDir, "data")
+	_, err := os.Stat(exportDataDir)
```

Comment:
Simplified the `discoverFilesToImport()` implementation to read the list from dataFileDescriptor.json--instead of relying on some arbitrary naming convention.
```
func discoverFilesToImport() []*ImportFileTask {
	result := []*ImportFileTask{}

	for i, fileEntry := range dataFileDescriptor.DataFileList {
		task := &ImportFileTask{
			ID:        i,
			FilePath:  fileEntry.FilePath,
			TableName: fileEntry.TableName,
		}
		result = append(result, task)
	}
	return result
}
```